apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -0,0 +1,392 @@
+      const aiInfo = aiEvaluations.find(
+        item => item.learning_goal_id === learningGoalId
+      );
+      return aiInfo;
+    } else {
+      return null;
+    }
+  };
+
+  const aiUnderstanding = getAiUnderstanding(
+    learningGoals[currentLearningGoal].id
+  );
+  const aiConfidence = getAiConfidence(learningGoals[currentLearningGoal].id);
+  const aiEvalInfo = getAiInfo(learningGoals[currentLearningGoal].id);
+
+  const autosave = () => {
+    setAutosaved(false);
+    setIsAutosaving(true);
+    setErrorAutosaving(false);
+    const bodyData = JSON.stringify({
+      studentId: studentLevelInfo.user_id,
+      learningGoalId: learningGoals[currentLearningGoal].id,
+      feedback: teacherFeedback.current,
+      understanding: understandingLevel.current,
+    });
+    HttpClient.put(
+      `${base_teacher_evaluation_endpoint}/${learningGoalEval.id}`,
+      bodyData,
+      true,
+      {
+        'Content-Type': 'application/json',
+      }
+    )
+      .then(() => {
+        setIsAutosaving(false);
+        setAutosaved(true);
+        if (!feedbackAdded) {
+          setFeedbackAdded(true);
+        }
+      })
+      .catch(error => {
+        console.log(error);
+        setIsAutosaving(false);
+        setErrorAutosaving(true);
+      });
+    clearTimeout(autosaveTimer.current);
+  };
+
+  useEffect(() => {
+    if (studentLevelInfo && learningGoals[currentLearningGoal].id) {
+      const body = JSON.stringify({
+        userId: studentLevelInfo.user_id,
+        learningGoalId: learningGoals[currentLearningGoal].id,
+      });
+      HttpClient.post(
+        `${base_teacher_evaluation_endpoint}/get_or_create_evaluation`,
+        body,
+        true,
+        {
+          'Content-Type': 'application/json',
+        }
+      )
+        .then(response => response.json())
+        .then(json => {
+          setLearningGoalEval(json);
+          if (json.feedback) {
+            teacherFeedback.current = json.feedback;
+            setDisplayFeedback(teacherFeedback.current);
+          }
+          if (json.understanding >= 0 && json.understanding !== null) {
+            setDisplayUnderstanding(json.understanding);
+            understandingLevel.current = json.understanding;
+          }
+        })
+        .catch(error => console.log(error));
+    }
+  }, [studentLevelInfo, learningGoals, currentLearningGoal]);
+
+  // Callback to retrieve understanding data from EvidenceLevels
+  const radioButtonCallback = radioButtonData => {
+    analyticsReporter.sendEvent(EVENTS.TA_RUBRIC_EVIDENCE_LEVEL_SELECTED, {
+      ...(reportingData || {}),
+      learningGoalId: learningGoals[currentLearningGoal].id,
+      learningGoal: learningGoals[currentLearningGoal].learningGoal,
+      newlySelectedEvidenceLevel: radioButtonData,
+      previouslySelectedEvidenceLevel: understandingLevel.current,
+    });
+    setDisplayUnderstanding(radioButtonData);
+    understandingLevel.current = radioButtonData;
+    if (!isAutosaving) {
```

Comment:
What happens if someone clicks a radio button while an autosave is in progress? Do we just leave the new button selected without firing an autosave event? Could be problematic. 

Maybe consider disabling the radio buttons while a save request is in flight, or queuing up a new autosave after the current one if this happens?